frontend/src/components/Hints.tsx

```
@@ -0,0 +1,93 @@
+            {unlockedHints.map((hint) => (
+                <div key={hint.hintId} className={styles.hint}>
+                    <p>{hint.hint}</p>
+                </div>
+            ))}
+            {/* hide hints text button */}
+            {lockedHints.length > 0 &&
+                (isModalOpen ? (
+                    <div className={styles.modal}>
+                        <h4>Unlock Hint</h4>
+                        <div className={styles.buttons}>
+                            <button onClick={handleBuy}>
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)